[MON-21885] [Doc] Bad incrementation in VMware health (#2756)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/virtualization-vmware2-vcenter-generic.md

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> 

Reviewer: ```suggestion

VMWare provides cloud computing and virtualization software and services.
The Centreon Plugins and Monitoring Connectors rely on the Centreon VMWare Connector to request the vCenter API.
With this connector, Centreon can monitor VMs, Datastores, ESXs, Clusters, etc.

```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-plugin-Virtualization-Vmware2-Connector-Plugin
```

</TabItem>
</Tabs>

## Using the monitoring connector

### Using a host template provided by the connector

1. Log into Centreon and add a new host through **Configuration > Hosts**.
2. Fill the **Name**, **Alias** & **IP Address/DNS** fields according to your ressource settings.
3. Apply the **Virt-VMWare2-VCenter-custom** template to the host. A list of macros appears. Macros allow you to define how the connector will connect to the resource, and to customize the connector's behavior.
4. Fill in the macros you want. Some macros are mandatory.

| Macro                      | Description                                                                                           | Default value     | Mandatory   |
|:---------------------------|:------------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| CENTREONVMWAREPORT         | Connector port (default: 5700)                                                                        | 5700              |             |

> | CENTREONVMWARECONTAINER    | Container to use (it depends of the connector configuration)                                          | default           |             |

Reviewer: ```suggestion
| CENTREONVMWARECONTAINER    | Container to use (it depends on the connector's configuration)                                          | default           |             |
```

| CRITICALTOTALALARMCRITICAL | Critical threshold                                                                                                                                                          |                             |             |
| EXTRAOPTIONS               | Any extra option you may want to add to the command (e.g. a --verbose flag). All options are listed [here](#available-options)                                                                         | --verbose --ignore-warn-msg |             |

</TabItem>
<TabItem value="Datastore-Io-Global" label="Datastore-Io-Global">

| Macro              | Description                                                                                                                                                | Default value     | Mandatory   |
|:-------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER             | datastore name to list                                                                                                                                     | .*                |             |

Reviewer: ```suggestion
| FILTER             | The connector will only take into account alerts coming from the datastores listed here                                                                                                                            | .*                |             |
```

| WARNINGWRITE       | Warning threshold                                                                                                                                          |                   |             |
| CRITICALWRITE      | Critical threshold                                                                                                                                         |                   |             |
| EXTRAOPTIONS       | Any extra option you may want to add to the command (e.g. a --verbose flag). All options are listed [here](#available-options)                                                        | --verbose         |             |

</TabItem>
<TabItem value="Datastore-Iops-Global" label="Datastore-Iops-Global">

| Macro              | Description                                                                                                                                                | Default value     | Mandatory   |
|:-------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER             | datastore name to list                                                                                                                                     | .*                |             |

Reviewer: ```suggestion
| FILTER             | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                   | .*                |             |
```

| WARNINGWRITEVM     | Thresholds                                                                                                                                                 |                   |             |
| CRITICALWRITEVM    | Thresholds                                                                                                                                                 |                   |             |
| EXTRAOPTIONS       | Any extra option you may want to add to the command (e.g. a --verbose flag). All options are listed [here](#available-options)                                                        | --verbose         |             |

</TabItem>
<TabItem value="Datastore-Snapshots-Global" label="Datastore-Snapshots-Global">

| Macro            | Description                                                                                                                                                | Default value     | Mandatory   |
|:-----------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER           | datastore name to list                                                                                                                                     | .*                |             |

Reviewer: ```suggestion
| FILTER           | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                    | .*                |             |
```

| WARNINGTOTAL     | Warning threshold                                                                                                                                          |                   |             |
| CRITICALTOTAL    | Critical threshold                                                                                                                                         |                   |             |
| EXTRAOPTIONS     | Any extra option you may want to add to the command (e.g. a --verbose flag). All options are listed [here](#available-options)                                                        | --verbose         |             |

</TabItem>
<TabItem value="Datastore-Usage-Global" label="Datastore-Usage-Global">

| Macro               | Description                                                                                                                                                | Default value     | Mandatory   |
|:--------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER              | datastore name to list                                                                                                                                     | .*                |             |

Reviewer: ```suggestion
| FILTER              | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                 | .*                |             |
```

| WARNINGUSAGE        | Thresholds. : 'usage' (B), 'usage-free' (B), 'usage-prct' (%), 'provisioned'                                                                               | 80                |             |
| CRITICALUSAGE       | Thresholds. : 'usage' (B), 'usage-free' (B), 'usage-prct' (%), 'provisioned'                                                                               | 90                |             |
| EXTRAOPTIONS        | Any extra option you may want to add to the command (e.g. a --verbose flag). All options are listed [here](#available-options)                                                        | --verbose         |             |

</TabItem>
<TabItem value="Datastore-Vm-Count-Global" label="Datastore-Vm-Count-Global">

| Macro                  | Description                                                                                                                                                | Default value     | Mandatory   |
|:-----------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | FILTER                 | datastore name to check                                                                                                                                    | .*                |             |

Reviewer: ```suggestion
| FILTER                 | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                   | .*                |             |
```

| --connector-port                           | Connector port (default: 5700).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |

> | --container                                | Container to use (it depends of the connector configuration).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |

Reviewer: ```suggestion
| --container                                | Container to use (it depends on the connector's configuration).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
```

| --warning-*            | Warning threshold. Can be: 'total-alarm-warning', 'total-alarm-critical'.                                                                                                                                                                     |
| --critical-*           | Critical threshold. Can be: 'total-alarm-warning', 'total-alarm-critical'.                                                                                                                                                                    |

</TabItem>
<TabItem value="Datastore-Io-Global" label="Datastore-Io-Global">

| Option             | Description                                                                                                                                                  |
|:-------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | --datastore-name   | datastore name to list.                                                                                                                                      |

Reviewer: ```suggestion
| --datastore-name   | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                    |
```

| --warning-status   | Define the conditions to match for the status to be WARNING (Default: ''). You can use the following variables: %{accessible}                                |
| --critical-status  | Define the conditions to match for the status to be CRITICAL (Default: ''). You can use the following variables: %{accessible}                               |
| --warning-*        | Warning threshold. Can be: 'total-read', 'total-write', 'read', 'write'.                                                                                     |
| --critical-*       | Critical threshold. Can be: 'total-read', 'total-write', 'read', 'write'.                                                                                    |

</TabItem>
<TabItem value="Datastore-Iops-Global" label="Datastore-Iops-Global">

| Option                   | Description                                                                                                                                                  |
|:-------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | --datastore-name         | datastore name to list.                                                                                                                                      |

Reviewer: ```suggestion
| --datastore-name         | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                    |
```

| --warning-status         | Define the conditions to match for the status to be WARNING (Default: ''). You can use the following variables: %{accessible}                                |
| --critical-status        | Define the conditions to match for the status to be CRITICAL (Default: ''). You can use the following variables: %{accessible}                               |
| --warning-* --critical-* | Thresholds. Can be: 'read-total', 'write-total', 'read', 'write', 'read-vm', 'write-vm'.                                                                     |

</TabItem>
<TabItem value="Datastore-Snapshots-Global" label="Datastore-Snapshots-Global">

| Option             | Description                                                                                                                                                  |
|:-------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | --datastore-name   | datastore name to list.                                                                                                                                      |

Reviewer: ```suggestion
| --datastore-name   | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                    |
```

| --warning-status   | Define the conditions to match for the status to be WARNING (Default: ''). You can use the following variables: %{accessible}                                |
| --critical-status  | Define the conditions to match for the status to be CRITICAL (Default: ''). You can use the following variables: %{accessible}                               |
| --warning-*        | Warning threshold. Can be: 'total', 'snapshot'.                                                                                                              |
| --critical-*       | Critical threshold. Can be: 'total', 'snapshot'.                                                                                                             |

</TabItem>
<TabItem value="Datastore-Usage-Global" label="Datastore-Usage-Global">

| Option                   | Description                                                                                                                                                  |
|:-------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | --datastore-name         | datastore name to list.                                                                                                                                      |

Reviewer: ```suggestion
| --datastore-name         | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                     |
```

| --warning-status         | Define the conditions to match for the status to be WARNING (Default: ''). You can use the following variables: %{accessible}                                |
| --critical-status        | Define the conditions to match for the status to be CRITICAL (Default: ''). You can use the following variables: %{accessible}                               |
| --warning-* --critical-* | Thresholds. Can be: Can be: 'usage' (B), 'usage-free' (B), 'usage-prct' (%), 'provisioned'.                                                                  |

</TabItem>
<TabItem value="Datastore-Vm-Count-Global" label="Datastore-Vm-Count-Global">

| Option             | Description                                                                                                                                                  |
|:-------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | --datastore-name   | datastore name to check.                                                                                                                                     |

Reviewer: ```suggestion
| --datastore-name   | The connector will only take into account alerts coming from the datastores listed here                                                                                                                                    |
```

| --esx-hostname         | ESX hostname to check. If not set, we check all ESX.                                                                                                                                                                                          |
| --filter               | Datacenter is a regexp.                                                                                                                                                                                                                       |
| --scope-datacenter     | Search in following datacenter(s) (can be a regexp).                                                                                                                                                                                          |
| --scope-cluster        | Search in following cluster(s) (can be a regexp).                                                                                                                                                                                             |

> | --filter-time          | Don't check alarm older (value in seconds).                                                                                                                                                                                                   |

Reviewer: ```suggestion
| --filter-time          | The connector will ignore any alert older than the time period specified here (in seconds).                                                                                                                                                                                             |
```
